apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.ListBuckets.cs

```diff
@@ -26,7 +26,8 @@ public abstract partial class StorageClient
         /// </summary>
         /// <remarks>
         /// This method fetches the buckets lazily, making requests to the underlying service
-        /// for a page of results at a time, as required.
+        /// for a page of results at a time, as required. No network requests are made until the returned sequence
+        /// is enumerated, so any exception due to an invalid request will be deferred until that time.
```

Comment:
I copied the paragraph:

> This method fetches the buckets lazily, making requests to the underlying service for a page of results at a time, as required. No network requests are made until the returned sequence is enumerated, so any exception due to an invalid request will be deferred until that time.

into [Hemingway app](http://hemingwayapp.com/), and it scores like this:

Readability
Grade 13
OK.  Aim for 9.

I also think we should aim for 9.  For many readers, English is not their first language, and for readers with a college education, simpler language will communicate more quickly.

One phrase I had to reread to parse correctly:
>  for a page of results at a time

At what time?  Oh, one *page at a time.*

Also,

> so any exception due to an invalid request will be deferred until that time

It took me a while to translate this sentence into specific advice for myself: when I try to pull exceptions out of the enumeration later, it may throw an exception.  I need to prepare to catch it.

I tried rewriting the paragraph for 5 minutes, but I'm still not satisfied with my results.  I can put some more time and thought into it and propose alternate text, if you like.

Reply:
This gets to grade 9, basically by splitting the new sentence:

>  This method fetches the buckets lazily, making requests to the underlying service for a page of results at a time, as required. No network requests are made until the returned sequence is enumerated. This means that any exception due to an invalid request will be deferred until that time. 

I'm personally okay with the "for a page at a time" part - I *suspect* the line break was working against you, and of course that wouldn't be in the docs the user is reading. I can't think of any alternatives.

In terms of the advice, we could add this to the end:

> Callers should be prepared for exceptions to be thrown while enumerating the results. In addition to failures due to invalid requests, network or service failures can cause exceptions even after the first results have been returned.

I wouldn't want users thinking that it's only retrieving the *first* item that can cause a problem, even if that's the most likely place.

Reply:
With the addendum, I like it!

Reply:
Great! Will fix storage and amend this first commit, then roll out a second commit for BigQuery, then merge. Yay.